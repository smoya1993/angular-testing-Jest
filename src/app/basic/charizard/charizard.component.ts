import { Component, OnInit } from '@angular/core';
import { PokemonService } from '../services/pokemon.service';
import { Pokemon } from '../interfaces';

@Component({
  selector: 'app-charizard',
  templateUrl: './charizard.component.html',
  styleUrls: ['./charizard.component.scss']
})
export class CharizardComponent implements OnInit {

  public charizard?: Pokemon;

  constructor( private pokemonService: PokemonService ) { }

  ngOnInit(): void {

    this.pokemonService.getPokemon(6)
      .subscribe( pokemon => {
        this.charizard = pokemon;
        // console.log(pokemon);
      });

  }

}
