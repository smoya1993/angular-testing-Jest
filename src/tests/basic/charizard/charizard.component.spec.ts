import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CharizardComponent } from '../../../app/basic/charizard/charizard.component';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { PokemonService } from '../../../app/basic/services/pokemon.service';


describe('CharizardComponent', () => {
  let component: CharizardComponent;
  let fixture: ComponentFixture<CharizardComponent>;
  let compiled: HTMLElement;
  let service: PokemonService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ CharizardComponent ],
      imports: [ HttpClientTestingModule ],
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CharizardComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
    service = TestBed.inject( PokemonService );
    httpMock = TestBed.inject( HttpTestingController );
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should have to match snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();    
  });

  test('should show loading', () => {
    const h2 = compiled.querySelector('h2');
    expect( h2?.textContent ).toContain('Loading...');
  });

  test('should load information immediately', () => {

    const dummyPokemon = {
      name: 'charizardo1!!',
      sprites: {
        front_default: 'https://charizard.com/sprite.png'
      }
    };

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6');
    expect( request.request.method ).toBe('GET');
     request.flush( dummyPokemon );

    fixture.detectChanges();
     //console.log(compiled.innerHTML);
    const h3  = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect( h3?.textContent?.toLocaleLowerCase() ).toContain( dummyPokemon.name.toLocaleLowerCase() );
    expect( img?.src ).toBe( dummyPokemon.sprites.front_default );
    expect( img?.alt ).toBe( dummyPokemon.name );
  });
});
