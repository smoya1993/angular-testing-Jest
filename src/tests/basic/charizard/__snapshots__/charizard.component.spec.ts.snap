// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharizardComponent should have to match snapshot 1`] = `
"<h1>Pok√©mon</h1><div><h2>Loading...</h2></div><!--bindings={
  "ng-reflect-ng-if-else": "[object Object]"
}--><!--container-->"
`;
